### PR TITLE
parameterize simplification of line features in read_shp

### PR DESCRIPTION
**Pull request by chrisnatali**
I needed the segments of multi-segment line features to be added as edges, so I added a simplify parameter to read_shp (defaults to True so it's back compatible).  A caveat is that attributes associated with the feature are repeated for each segment.  Thought someone else might find this useful.

### CODE REVIEW

**Comment by chebee7i** [comment]
Seems helpful. Could you add a unit test that covers this case? You should be able to modify one of the tests that appears in `networkx/readwrite/tests/test_shp.py`.

File: networkx/readwrite/nx_shp.py [hunk 1]
```diff
@@ -38,6 +38,11 @@ def read_shp(path):
     path : file or string
        File, directory, or filename to read.
 
+    simplify:  Simplify line geometries to start and end coordinates
+        If False, and line feature geometry has multiple segments, the 
+        attributes for that feature will be repeated for each edge comprising
+        that feature
+
```
Comment (chebee7i) [inline networkx/readwrite/nx_shp.py]: It should be:
```
`simplify : bool
    If ``True``, simplify line geometries to the start and end 
    coordinates. If ``False``... comprising that feature.
```

File: networkx/readwrite/nx_shp.py [hunk 2]
```diff
@@ -73,8 +78,12 @@ def read_shp(path):
                 attributes["Wkb"] = g.ExportToWkb()
                 attributes["Wkt"] = g.ExportToWkt()
                 attributes["Json"] = g.ExportToJson()
-                last = g.GetPointCount() - 1
-                net.add_edge(g.GetPoint_2D(0), g.GetPoint_2D(last), attributes)
+                if simplify:
+                    last = g.GetPointCount() - 1
+                    net.add_edge(g.GetPoint_2D(0), g.GetPoint_2D(last), attributes)
+                else:
+                    for i in range(0, g.GetPointCount() - 1):
```
Comment (chebee7i) [inline networkx/readwrite/nx_shp.py]: Might as well change this at the same time: The start integer 0 is typically not included in `range`. So just `range(g.GetPointCount() - 1)` should do.

Comment (SanketDG) [inline networkx/readwrite/nx_shp.py]: The `last` variable should be declared after line 80 where it previously was. Then line 85 should be changed to 
```python
for i in range(last):

Don't understand why variable `last` is inside the `if` block.

**Comment by chrisnatali** [comment]
Thanks for the feedback.  Addressed comments.  Also for the ``simplify=False`` case it seemed consistent to "segmentize" the "Wkb", "Wkt" and "Json" attributes.

**Pull request merged**